bayes_opt/domain_reduction.py

```diff
@@ -97,15 +97,23 @@ def _update(self, target_space: TargetSpace) -> None:
         self.r = self.contraction_rate * self.r
 
     def _trim(self, new_bounds: np.array, global_bounds: np.array) -> np.array:
-        for i, variable in enumerate(new_bounds):
-            if variable[0] < global_bounds[i, 0]:
-                variable[0] = global_bounds[i, 0]
-            if variable[1] > global_bounds[i, 1]:
-                variable[1] = global_bounds[i, 1]
+        """ Tests new_bounds to enforce global_bounds and minimum_window."""
+        # sort the order of the bounds for each parameter
+        for i, pbounds in enumerate(new_bounds):
+            if pbounds[0] > pbounds[1]:
+                new_bounds[i, 0] = pbounds[1]
+                new_bounds[i, 1] = pbounds[0]
```

Comment:
I think this can malfunction since iterating over an array (`__next__`) returns a pointer to the original object's data.

```python
import numpy as np
new_bounds = np.array([[2., 1.], [4., 3.]])
for i, pbounds in enumerate(new_bounds):
    if pbounds[0] > pbounds[1]:
        new_bounds[i, 0] = pbounds[1]
        new_bounds[i, 1] = pbounds[0]
```
Out:
```
array([[1., 1.],
       [3., 3.]])
```

Reply:
I had a suspicion this could be a problem, but that's how it was previously implemented, so I left it as-is. Anyways, this can be handled by np.sort(pbounds) instead. I'll make the change.

Reply:
Yeah, you're right, it must've slipped by the last time. Thanks for handling it!

Reply:
done